File: app/src/main/java/com/github/gfx/helium/model/EpitomeEntry.java
package com.github.gfx.helium.model;

import com.google.gson.FieldNamingPolicy;

import com.github.gfx.static_gson.annotation.JsonSerializable;

import org.threeten.bp.ZonedDateTime;

import java.util.List;

@JsonSerializable(
        fieldNamingPolicy = FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES
)
public class EpitomeEntry {

    final static String SCHEME_GISTS = "gists";

    public String id;

    public String scheme;

    public String title;

    public int views;

    public String epitomeUrl;

    public String upstreamUrl;

    public String publishedAt;

    public List<Gist> gists;

    public ZonedDateTime getTimestamp() {
        return ZonedDateTime.parse(publishedAt);
    }

    public boolean hasKnownScheme() {
        return isGists();
    }

    public boolean isGists() {
        return SCHEME_GISTS.equals(scheme);
    }

    @JsonSerializable
    public static class Gist {

        public String content;
    }
}

File: app/src/main/java/com/github/gfx/helium/model/EpitomeEntry.kt
package com.github.gfx.helium.model

import com.google.gson.annotations.SerializedName

import org.threeten.bp.ZonedDateTime

class EpitomeEntry {

    @SerializedName("id")
    lateinit var id: String

    @SerializedName("scheme")
    lateinit var scheme: String

    @SerializedName("title")
    lateinit var title: String

    @SerializedName("views")
    var views: Int  = 0

    @SerializedName("epitome_url")
    lateinit var epitomeUrl: String

    @SerializedName("upstream_url")
    lateinit var upstreamUrl: String

    @SerializedName("published_at")
    lateinit var publishedAt: String

    @SerializedName("gists")
    var gists: List<Gist>? = null

    var timestamp: ZonedDateTime? = null
        get() = ZonedDateTime.parse(publishedAt)

    fun hasKnownScheme(): Boolean {
        return isGists
    }

    val isGists: Boolean
        get() = SCHEME_GISTS == scheme

    class Gist {

        @SerializedName("content")
        lateinit var content: String
    }

    companion object {

        internal val SCHEME_GISTS = "gists"
    }
}
